planning_interface/move_group_interface/src/move_group.cpp

```diff
@@ -689,6 +689,38 @@ class MoveGroup::MoveGroupImpl
     }
   }
 
+  bool validatePlan(const Plan &plan)
+  {
+    robot_state::RobotStatePtr current_state;
+    if (!getCurrentState(current_state))
+      return false;
+    if (plan.trajectory_.joint_trajectory.points.empty())
+      return true;
+
+    const trajectory_msgs::JointTrajectory &trajectory = plan.trajectory_.joint_trajectory;
+    const std::vector<double> &positions = trajectory.points.front().positions;
+    std::size_t n = trajectory.joint_names.size();
+    if (positions.size() != n)
+      return false;
+
+    for (std::size_t i = 0; i < n; ++i)
+    {
+      const robot_model::JointModel *jm = robot_model_->getJointModel(trajectory.joint_names[i]);
+      if (!jm)
+      {
+        ROS_ERROR_STREAM("Unknown joint in trajectory: " << trajectory.joint_names[i]);
+        return false;
+      }
+      // TODO: check multi-DoF joints
+      if (fabs(current_state->getJointPositions(jm)[0] - positions[i]) < std::numeric_limits<float>::epsilon())
```

Comment:
Switching to the machine epsilon check, I inserted a new bug: 
The comparison operator needs to be inversed.